include/mbgl/shaders/mtl/common.hpp

```diff
@@ -10,10 +10,20 @@ constexpr auto prelude = R"(
 #include <metal_stdlib>
 using namespace metal;
 
+// The maximum allowed miter limit is 2.0 at the moment. the extrude normal is stored
+// in a byte (-128..127). We scale regular normals up to length 63, but there are also
+// "special" normals that have a bigger length (of up to 126 in this case).
+#define LINE_NORMAL_SCALE (1.0 / (127 / 2))
+
+// The attribute conveying progress along a line is scaled to [0, 2^15).
+#define MAX_LINE_DISTANCE 32767.0
+
+#define SDF_PX 8.0
+
+// OpenGL `mod` is `x-y*floor(x/y)` where `floor` rounds down.
+// Metal `fmod` is `x-y*trunc(x/y)` where `trunc` rounds toward zero.
 template <typename T1, typename T2>
-inline auto mod(T1 x, T2 y) -> decltype(x - y * floor(x/y)) {
-    return x - y * metal::floor(x/y);
-}
+inline auto glMod(T1 x, T2 y) -> auto { return x - y * metal::floor(x/y); }
```

Comment:
Changing the name to `glMod` will break new PRs that uses `mod` and not yet merged. I am ok with `mod` name, considering that is not a conflict with metal name, which is `fmod`. `mod` is consistent between Metal and GL.

Reply:
IMO `mod` and `fmod` having different behavior is counterintuitive and should be avoided.

Reply:
Yes, but `mod` and `fmod` have different names, so different behaviour.

Reply:
Anyway, not so important, the PR is approved from my side.

Reply:
@louwers @mwilsnd Do you have an opinion?

Reply:
It looks like the difference is between the behavior of `floor()` and `trunc()`? I think being clear is a pro here, but perhaps being more explicit, with a comment attached.

Reading the metal specs, it's the rounding mode that is different - "negative infinity rounding mode" for `floor()` and "round toward zero rounding mode" for Metal. I'd include that info in a comment and make it clear that GL uses the `floor()` variant. The name doesn't matter too much, so long as it's clear what's happening and why.

Reply:
There's more context in the code.

https://github.com/maplibre/maplibre-native/pull/1618/files/8c46ea2141c81c0c3e5e8662f51a5e3efc8b73bc#diff-eb8e1270fc899b64368fc68f795b7bf297ef32892f1535fbd5ac8ce69320716c

Reply:
Yeah, then I'm fine with it being `glMod`. Merges for in-flight work should be trivial for this.